Venmo/src/main/java/com/braintreepayments/api/VenmoClient.java

```diff
@@ -163,10 +169,14 @@ public void onResult(@Nullable final Configuration configuration, @Nullable Exce
                 }
 
                 String exceptionMessage = null;
+                if (!request.getFallbackToWeb()) {
+                     if (!deviceInspector.isVenmoAppSwitchAvailable(activity)) {
```

Comment:
I thought we were going to keep the logic that checks if the app is installed before doing the switch, even if the switch is via app links?

Reply:
Are you thinking for analytics purposes like sending a new event here? We wouldn't want to set the exception message if we are falling back to web. On like 182 we check the value of this message and callback with the exception if it exists. For the web fallback case, we should expect the exception to be null below.

Reply:
No not for analytics purposes. I thought the purpose of moving to universal links was to replace the URL scheme switch and hope for less drop-off with "unknown" errors that happen, not to remove the check for if the app is installed entirely.

Reply:
We shouldn't need the app installed check for this flow since we rely on the app links URL + OS to make this determination. We don't fallback to the old Intent flow if the app is not installed. In both cases for app links we hit `https://venmo.com/go/checkout` and the OS does the checks. I am not sure why we'd still want to do the check additionally.

Reply:
Gotcha - I get that the check is not necessary to perform the switch. I just thought we were planning to keep it until the next major version. 

Keeping it would let us more accurately compare the performance of intents vs app link switches at the OS level.

Reply:
^ This doesn't have to be a blocking comment, but this might come up w/ the Venmo team again

Reply:
🤔 I may be completely misunderstanding what you are saying, so if I am happy to talk irl, but it doesn't look like we send anything in the check for `isVenmoAppSwitchAvailable` ([it calls this method under the hood](https://github.com/braintree/braintree_android/blob/fdd6bdc23e5a9c076ff800939d79096c9f64a0dc/BraintreeCore/src/main/java/com/braintreepayments/api/DeviceInspector.kt#L58-L64)) that would allow us to compare performance. Again, maybe I am completely misunderstanding what you're saying. 😅

Reply:
Totally down to chat IRL! Also don't let this convo block the PR.

What I mean is: I imagine we'll want to be comparing "conversion" rate for the intent flow versus the app link flow, so comparing # start to # end events. That comparison is more apples-to-apples if the intentional web fallback is removed from the equation, if both situations check for the app being installed before attempting to switch.

Let's say the app link flow (which currently doesn't check for app install) shows us awful conversion rates. We won't know if that's because the web fallback conversion stinks, _or_ the if the general app <--> app switch behavior on the app link rails is problematic and inconsistent.

Reply:
Apologies for the confusion! More of a general ❓

Reply:
To close this thread - we're going forward with the logic, as-is! For intentionally falling back to the web flow in the cases of no app installed.

Confirmed with Jax IRL and also with the Venmo product team.

Reply:
@jaxdesmarais Should I leave this thread unresolved until we can lang on an analytics piece? Also happy to 🏈 punt on the analytics piece here for a future PR too.

IRL we discussed that knowing what portion of users don't have the app installed will allow us to determine what % of users are being sent to the web fallback, or not.

Reply:
Do with think it'd be helpful to add analytics in the case of web fallback + app not installed? We could also add this in a future PR if we want more granular data on the flow. 🤔

Reply:
LOL, we asked the same question. I think I lean towards adding it in this PR as it could give us some insight between the two flows (intent vs app links) as well as the performance of app not installed + web fallback vs app installed + web fallback. The question is really do we want to send it for _all_ flows or just not installed + web fallback? What do you think?